Automattic/jetpack
# Update/boost score bar to react component

This PR adds a Jetpack Boost Speed Score Bar component to the components package so it can be used in both the Jetpack Boost and Jetpack plugin

## Proposed changes:
 * Add a Jetpack Boost Speed Score Bar component
 * Make it a storybook component
 * Add tests

## Jetpack product discussion

P2: pbNhbs-4T7-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Checkout these changes
2. Start your local Jetpack environment
3. Run `jetpack install plugins/boost && jetpack build plugins/boost` and have `jetpack watch plugins/boost` going since that is where we will be testing this component
4. Install the Boost plugin and optionally purchase a plan for it
5. Go to `/wp-admin/admin.php?page=jetpack-boost`
6. Add the following imports
```svelte
import { BoostScoreBar } from '@automattic/jetpack-components';
import ReactComponent from '../../../elements/ReactComponent.svelte';
```
7. Go to the file `projects/plugins/boost/app/assets/src/js/pages/settings/sections/Score.svelte` and after line 178, insert the following lines of code
```svelte
        <br />
        <br />

        <ReactComponent
            this={BoostScoreBar}
            prevScore={scores.noBoost?.mobile}
            score={scores.current.mobile}
            active={siteIsOnline}
            {isLoading}
            {showPrevScores}
            scoreBarType="mobile"
            noBoostScoreTooltip={__( 'Your mobile score without Boost', 'jetpack-boost' )}
        />

        <ReactComponent
            this={BoostScoreBar}
            prevScore={scores.noBoost?.desktop}
            score={scores.current.desktop}
            active={siteIsOnline}
            {isLoading}
            {showPrevScores}
            scoreBarType="desktop"
            noBoostScoreTooltip={__( 'Your desktop score without Boost', 'jetpack-boost' )}
        />
```
8. Refresh the page and ensure the two bars on the bottom look and act the same as the bars on the top
![image](https://user-images.githubusercontent.com/65001528/231285694-7c6ce7e8-5508-43c8-a9b5-2b2cccaf350d.png)
9. Test out the Refresh button and the tooltip when you hover over the previous score ( to see this, you'll need to run a speed score, turn on some boost features, and then run it again )
10. To see the storybook component, go to `projects/js-packages/storybook` and run `npm run storybook:build && npm run storybook:dev`. The link will be provided once the build is done, but it should be http://localhost:50240/?path=/story/js-packages-components-boost-score-bar--default
11. To run the tests manually, go to `projects/js-packages/components` and run `npm run test components/boost-score-bar`
